### PR TITLE
schedule: change merge match peers strategy

### PR DESCRIPTION
**Pull request by Connor1996**
### What problem does this PR solve? 
There is a case that a follower is added and transfer leader to it, and the apply process of it is slow so leader regards it as a voter but actually it is still a learner. Once that, the follower can't be a leader, but the old leader can't know that so there is no leader to serve for a while.

### What is changed and how it works?
change merge match peers strategy, transfer leader to the first added follower if there is no intersection store, and remove one peer after adding one peer.

### Check List 
 - Unit test

### CODE REVIEW

**Comment by nolouch** [comment]
@Connor1996  CI failed.

File: server/schedule/operator.go [hunk 1]
```diff
@@ -521,21 +526,25 @@ func matchPeerSteps(cluster Cluster, source *core.RegionInfo, target *core.Regio
...
 	intersection := getIntersectionStores(sourcePeers, targetPeers)
 	leaderID := source.GetLeader().GetStoreId()
-	isFound := false
 	for _, storeID := range intersection {
 		if storeID == leaderID {
-			isFound = true
+			targetLeader = 0
 			break
 		}
+		targetLeader = storeID
```
Comment (disksing) [inline server/schedule/operator.go]: It seems the `targetLeader` will be the last store in `intersection` instead of _first added follower_.

Comment (Connor1996) [reply server/schedule/operator.go]: yeah, if there is an intersection, no need to transfer to first added follower

**Comment by Connor1996** [comment]
@nolouch Because `range` cannot keep same order every time. It is a little hard to write this test case, and I am trying to figure it out.

**Comment by disksing** [comment]
@Connor1996 Any update? Maybe you can make sure regions have only 1 intersection.

File: server/schedule/operator.go [hunk 2]
```diff
@@ -492,68 +493,104 @@ func CreateMergeRegionOperator(desc string, cluster Cluster, source *core.Region
...
+
+		// the leader should be the last to remove
+		if targetLeader != 0 && peer.GetStoreId() == leaderID {
+			continue
+		}
+
+		steps = append(steps, toAdds[index]...)
 		steps = append(steps, RemovePeer{FromStore: peer.GetStoreId()})
 		kind |= OpRegion
+		index++
```
Comment (zhangjinpeng1987) [inline server/schedule/operator.go]: Please check if index == len(toAdds) after this loop.

**Pull request merged**